src/main/java/io/supertokens/totp/Totp.java

```diff
@@ -139,8 +139,8 @@ private static void checkAndStoreCode(Main main, TOTPStorage totpStorage, String
                         if (now - latestInvalidCodeCreatedTime < rateLimitResetTimeInMs) {
                             // Less than rateLimitResetTimeInMs (default = 15 mins) time has elasped since
                             // the last invalid code:
-                            throw new StorageTransactionLogicException(
-                                    new LimitReachedException(rateLimitResetTimeInMs / 1000));
+                            int timeLeftMs = (int) (rateLimitResetTimeInMs - (now - latestInvalidCodeCreatedTime));
```

Comment:
- should be long
- have you tested that this value is correct?

Reply:
done.